src/metabase/db.clj

```diff
@@ -203,6 +203,23 @@
          ^Database       database       (.findCorrectDatabaseImplementation (database-factory) liquibase-conn)]
      (Liquibase. changelog-file (ClassLoaderResourceAccessor.) database))))
 
+(defn consolidate-liquibase-changesets
+  "Consolidate all previous DB migrations so they came from single file.
+   Previous migrations where stored in many small files which added seconds
+   per file to the startup time because liquibase was checking the jar
+   signature for each file. This function is required to correct the liquibase
+   tables to reflect that these migrations where moved to a single file.
+
+   see https://github.com/metabase/metabase/issues/3715"
+  [conn]
+  (let [fresh-install? (jdbc/with-db-metadata [meta (jdbc-details)] ;; don't migrate on fresh install
+                         (empty? (jdbc/metadata-query (.getTables meta nil nil "DATABASECHANGELOG" (into-array String ["TABLE"])))))
+        query (if (= (db-type) :h2)
+                "UPDATE DATABASECHANGELOG SET FILENAME = ?"
+                "UPDATE databasechangelog SET filename = ?")]
+    (when (not fresh-install?)
```

Comment:
I would vote for `when-not` here:

```clojure
(when-not fresh-install?
  ...)
```

See https://github.com/bbatsov/clojure-style-guide#when-not